ngraph/test/op_eval/interpolate.cpp

```diff
@@ -0,0 +1,501 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/interpolate.hpp"
+#include "ngraph/runtime/host_tensor.hpp"
+#include "ngraph/validation_util.hpp"
+#include "runtime/backend.hpp"
+#include "util/all_close_f.hpp"
+#include "util/test_tools.hpp"
+#include "util/type_prop.hpp"
+
+using namespace ngraph;
+
+using InterpolateMode = op::v4::Interpolate::InterpolateMode;
+using CoordinateTransformMode = op::v4::Interpolate::CoordinateTransformMode;
+using Nearest_mode = op::v4::Interpolate::NearestMode;
+using InterpolateAttrs = op::v4::Interpolate::InterpolateAttrs;
+using ShapeCalcMode = op::v4::Interpolate::ShapeCalcMode;
+
+// All examples are from ONNX Resize-11 documentation
+// (see https://github.com/onnx/onnx/blob/master/docs/Operators.md).
+TEST(op_eval, interpolate_v4_cubic)
+{
+    auto data_shape = Shape{1, 1, 4, 4};
+
+    struct ShapesAndAttrs
+    {
+        std::vector<int64_t> spatial_shape;
+        Shape out_shape;
+        std::vector<float> scales_data;
+        CoordinateTransformMode transform_mode;
+        ShapeCalcMode shape_calculation_mode;
+    };
+
+    std::vector<ShapesAndAttrs> shapes_and_attrs = {
+        // resize_downsample_scales_cubic:
+        ShapesAndAttrs{{3, 3},
+                       Shape{1, 1, 3, 3},
+                       {0.8f, 0.8f},
+                       CoordinateTransformMode::half_pixel,
+                       ShapeCalcMode::scales},
+//         // resize_downsample_sizes_cubic:
+//         ShapesAndAttrs{{3, 3},
```

Comment:
Why did you comment this code?

Reply:
To debug `cubic` mode for 32-bit Ubuntu 18.